feat: DIA-2142: Use the new Button component in LSO apps and libraries

nick-skriabin opened the pull request:
This pull request focuses on one main area: refactoring the `Button` component to use a shared library. Changes in this PR spread across entire LSO - LSO React App, Data Manager and Editor. Be careful while reviewing.

Below is a summary of the most important changes:

### Button Component Refactoring

* Removed the custom `Button` component and its associated styles (`Button.jsx` and `Button.scss`) in favor of the shared `@humansignal/ui` library. This simplifies maintenance and ensures consistency across projects. [[1]](diffhunk://#diff-582827abc39080c4ae2aa299d75dcd835968549046e0a5f9a3279dc28827d427L1-L81) [[2]](diffhunk://#diff-eeb341ad3d194149f2a869f7ee543793eedb45c76212f045306160966666c779L1-L360)

* Updated all references to the old `Button` component across multiple files (e.g., `Error.jsx`, `Form.jsx`, `HeidiTip.tsx`, `Modal.jsx`, `ModalPopup.jsx`) to use the new `Button` from `@humansignal/ui`. Adjusted props and styles to match the new component's API. [[1]](diffhunk://#diff-f9e26be58c78a463d7bb7efcf66dfa571e9b3b819647fe736c683d4a177496b7L6-R6) [[2]](diffhunk://#diff-5e0cbad5e9c63daf36c8671ecd3969166f5812b795aac37834c1b1cedf90b90bL8-R8) [[3]](diffhunk://#diff-6c60f330dfea598fa1ef72af9e384b37745f2878caac5fabc4369a9c6a0c67b4L5-R5) [[4]](diffhunk://#diff-83ca9f15e39cf9246a473dabea8f46211cb3770f0f37a9bbf9e0026a3b80c358L8-R8) [[5]](diffhunk://#diff-6f384ee14bef129371794a8d7942469d1554f96a68efc23ff89e32a5781d6683L6-R6)

* Removed the `Button` export from the `index.js` file, as it is no longer part of the local codebase.

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

nick-skriabin committed:
Button WIP - Primary variant

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2142/button

nick-skriabin committed:
Finalize styles and variants

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2142/button

nick-skriabin committed:
Add `buttonVariant` helper function

- `buttonVariant` to apply button styles to different elements
- Example of `buttonVariant` usage with `a` example

nick-skriabin committed:
Proper link

nick-skriabin committed:
Fix link preview

nick-skriabin committed:
Add Button to export list, organize exports

nick-skriabin committed:
Start using new Button

nick-skriabin committed:
Remove text shadow from outlined button

nick-skriabin committed:
Merge branch 'fb-dia-2156/button' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Homepage fix button usage

nick-skriabin committed:
Replace "Create" button

nick-skriabin committed:
Add sizes and fix some UI bugs

nick-skriabin committed:
Fix button label paddings

nick-skriabin committed:
Merge branch 'fb-dia-2156/button' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Update buttons

nick-skriabin committed:
Fix sizing options

nick-skriabin committed:
Merge branch 'fb-dia-2156/button' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Update DM buttons

nick-skriabin committed:
Fix hover

nick-skriabin committed:
Merge branch 'fb-dia-2156/button' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Update buttons in project settings

nick-skriabin committed:
Update css formatting

nick-skriabin committed:
Merge branch 'fb-dia-2156/button' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Simplify button layout for complex scenarios

nick-skriabin committed:
Updating button layout

nick-skriabin committed:
Icons alignment

nick-skriabin committed:
Merge branch 'fb-dia-2156/button' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Updating buttons

nick-skriabin committed:
Fix button behavior when using outline/string look

nick-skriabin committed:
Merge branch 'fb-dia-2156/button' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Update hover behavior for outline/string looks

nick-skriabin committed:
Shadow behavior when focused/active

nick-skriabin committed:
Merge branch 'fb-dia-2156/button' into fb-dia-2142/new-button-lso

nick-skriabin committed:
More buttons

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2156/button

nick-skriabin committed:
Use raw values instead of hardcoded ones

nick-skriabin committed:
Update color variable

nick-skriabin committed:
Merge branch 'fb-dia-2156/button' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Remove LSO Button

nick-skriabin committed:
Replace buttons in Data Manager

nick-skriabin committed:
Update fonts

nick-skriabin committed:
Merge branch 'fb-dia-2156/button' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Updating Editor buttons

nick-skriabin committed:
Icon sizing

nick-skriabin committed:
Merge branch 'fb-dia-2156/button' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Update task navigation buttons

nick-skriabin committed:
Smaller text size

nick-skriabin committed:
Allow overriding `disabled` state when waiting

nick-skriabin committed:
Fix neutral button focus outline

nick-skriabin committed:
Merge branch 'fb-dia-2156/button' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Updating native buttons

nick-skriabin committed:
Fix Button usage in DM (caused error)

nick-skriabin committed:
Add ButtonGroup

nick-skriabin committed:
Merge branch 'fb-dia-2156/button' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Editor buttons

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Hotkeys

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Fix config labels

yyassi-heartex commented on web/apps/labelstudio/src/components/Form/Form.jsx:
```diff
@@ -524,7 +524,7 @@ Form.Builder = React.forwardRef(
         {props.autosubmit !== true && withActions === true && (
           <Form.Actions>
-            <Button type="submit" look="primary" style={{ width: 120 }}>
+            <Button type="submit" style={{ width: 120 }}>
```
does new button accept className? ideally here we should use something like `w-[120]` instead of inline styles

bmartel commented on web/apps/labelstudio/src/pages/Home/HomePage.tsx:
```diff
@@ -1,5 +1,5 @@
 import type { Page } from "../types/Page";
-import { Button } from "@humansignal/shad/components/ui/button";
+import { Button } from "@humansignal/ui";
 import { SimpleCard, Spinner } from "@humansignal/ui";
```
```suggestion
import { Button, SimpleCard, Spinner } from "@humansignal/ui";
```

bmartel commented on web/apps/labelstudio/src/pages/Organization/Models/ModelsPage.tsx:
```diff
@@ -1,8 +1,9 @@
-import { Button } from "apps/labelstudio/src/components";
 import type { Page } from "../../types/Page";
-import { Space } from "apps/labelstudio/src/components/Space/Space";
+import { Space } from "@humansignal/ui";
 import { Block } from "apps/labelstudio/src/utils/bem";
 import { EmptyList } from "./@components/EmptyList";
+import { Link } from "react-router-dom";
+import { buttonVariant } from "@humansignal/ui";
```
```suggestion
import { buttonVariant, Space } from "@humansignal/ui";
import { Block } from "apps/labelstudio/src/utils/bem";
import { EmptyList } from "./@components/EmptyList";
import { Link } from "react-router-dom";
```

yyassi-heartex commented on web/apps/labelstudio/src/pages/CreateProject/Config/Config.jsx:
```diff
@@ -533,7 +527,7 @@ const Configurator = ({
               </Block>
             )}
-            <Button look="primary" size="compact" style={{ width: 120 }} onClick={onSave} waiting={waiting}>
+            <Button size="small" style={{ width: 120 }} onClick={onSave} waiting={waiting}>
```
same thing here with the inline-style

yyassi-heartex commented on web/libs/editor/src/common/Hotkey/WithHotkey.tsx:
```diff
@@ -0,0 +1,36 @@
+    Children.only(children);
+
+    const { onClick, tooltip } = children.props;
+    const clone = cloneElement(children, { ref, tooltip: undefined, tooltipTheme: "dark" });
```
should we be defining the tooltip theme? how does this interact with dark-mode? cc @bmartel @ricardoantoniocm

ricardoantoniocm replied:
We should not hard code the theme into any component. The tokens should already take care of this. @nick-skriabin

yyassi-heartex commented on web/libs/editor/src/components/AnnotationsCarousel/AnnotationsCarousel.tsx:
```diff
@@ -86,26 +83,24 @@ export const AnnotationsCarousel = observer(({ store, annotationStore }: Annotat
+          </Button>
+          <Button
             disabled={isRightDisabled}
-            mod={{ right: true, disabled: isRightDisabled }}
             aria-label="Carousel right"
-            onClick={(e: MouseEvent) => !isRightDisabled && updatePosition(e, false)}
+            size="small"
+            variant="neutral"
+            onClick={(e) => !isRightDisabled && updatePosition(e, false)}
           >
-            <Elem name="arrow" mod={{ right: true }} tag={IconChevron} />
-          </Elem>
+            <IconChevron style={{ rotate: "90deg" }} />
```
now that the icon work is done we should be able to use `IconChevronLeft` and IconChevronRight` instead of rotating `IconChevron`

yyassi-heartex commented on web/libs/editor/src/components/App/App.jsx:
```diff
@@ -104,7 +104,7 @@ class App extends Component {
         <Block name="sub__result">You have completed all tasks in the queue!</Block>
         {store.taskHistory.length > 0 && (
-          <Button onClick={(e) => store.prevTask(e, true)} look="outlined" style={{ margin: "16px 0" }}>
+          <Button onClick={(e) => store.prevTask(e, true)} variant="neutral" style={{ margin: "16px 0" }}>
```
there is likely a tailwind class for this - probably something like `mx-0 my-[16]`

yyassi-heartex commented on web/libs/editor/src/components/BottomBar/BottomBar.scss:
```diff
@@ -22,6 +22,7 @@
     padding: 0 8px;
     align-items: center;
     box-sizing: border-box;
+    gap: 8px;
```
```suggestion
    gap: var(--spacing-small);
```

yyassi-heartex commented on web/libs/editor/src/components/CurrentEntity/Controls.jsx:
```diff
@@ -0,0 +1,101 @@
+    } else if (annotation.skipped) {
+      buttons.push(
+        <Elem name="skipped-info" key="skipped">
+          <IconBan color="#d00" /> Annotation skipped
```
we should look at applying a css class pased on the figma token - likely something `var(--color-negative-emphasis)` @ricardoantoniocm please confirm which color this should be

ricardoantoniocm replied:
The semantic variable for negative-looking icons should be `color-negative-icon`

yyassi-heartex commented on web/libs/editor/src/components/Entities/Entities.jsx:
```diff
@@ -0,0 +1,113 @@
+import { Dropdown } from "antd";
```
this is likely copied in from another spot - but antd instead of our Dropdown? 😢

nick-skriabin replied:
I'm not sure if this file even used anywhere since the Outliner is finally there without an FF

yyassi-heartex commented on web/libs/editor/src/components/Entities/Entities.jsx:
```diff
@@ -0,0 +1,113 @@
+                type="text"
+                aria-label="Delete All Regions"
+                icon={<IconTrash />}
+                style={{
+                  height: 36,
+                  width: 36,
+                  padding: 0,
+                }}
```
lets use our new tailwind/figma token css classes instead of inline-styles

yyassi-heartex commented on web/libs/editor/src/components/Entities/Entities.jsx:
```diff
@@ -0,0 +1,113 @@
+                  tag={Button}
+                  size="small"
+                  type="link"
+                  style={{ padding: 0 }}
```
`p-0` is probably what we're looking for here for className

yyassi-heartex commented on web/libs/editor/src/components/Entity/Entity.jsx:
```diff
@@ -0,0 +1,239 @@
+            placeholder="Meta Information"
+          />
+
+          <Button type="primary" htmlType="submit" style={{ marginRight: "0.5em" }}>
```
we're probably looking for something like `mr-3` here

yyassi-heartex commented on web/libs/editor/src/components/SidePanels/OutlinerPanel/ViewControls.tsx:
```diff
@@ -275,22 +272,22 @@ const ToggleRegionsVisibilityButton = observer<FC<ToggleRegionsVisibilityButton>
 
   return (
-    <Elem
+    <Button
```
since we switched to Button from Elem we no longer the `tag` to be passed in here

yyassi-heartex commented on web/libs/editor/src/core/Hotkey.ts:
```diff
@@ -225,6 +225,7 @@ export const Hotkey = (namespace = "global", description = "Hotkeys") => {
             func(...args);
           };
 
+          console.log("added", keyName, handler);
```
```suggestion
```

yyassi-heartex commented on web/libs/editor/src/stores/Annotation/Annotation.js:
```diff
@@ -880,7 +880,15 @@ const _Annotation = types
+          hotkeys.addKey(
+            node.hotkey,
+            () => {
+              console.log("fired", node.hotkey);
```
```suggestion
```

yyassi-heartex commented on web/libs/ui/src/lib/button/button.stories.tsx:
```diff
@@ -0,0 +1,190 @@
+import type { Meta, StoryObj } from "@storybook/react";
```
I love the way you did this ❤️ !

yyassi-heartex commented on web/libs/ui/src/lib/button/button.tsx:
```diff
@@ -0,0 +1,142 @@
+  leading?: ReactNode;
+  trailing?: ReactNode;
+  tooltip?: string;
+  tooltipTheme?: "light" | "dark";
```
minor nitpick here - if we can get this extracted to a type or an enum that we can export - this way when we want to pass it in - we can use that instead of typing the word in. reduces the risk of typos in the future

yyassi-heartex commented on web/libs/ui/src/lib/toast/toast.tsx:
```diff
@@ -85,9 +86,16 @@ export interface ToastActionProps extends ToastPrimitive.ToastActionProps {
+      size="small"
+      className={styles.toast__action}
+      onClick={onClose}
+      style={{ pointerEvents: "all" }}
```
i wouldnt be surprised if there is a tailwind class to do this

nick-skriabin replied:
surprisingly, there's only for `pointer-events-none` or `auto`, but no class for `all` and `pointer-evens` does not accept arbitrary values. luckily we can create utility classes

yyassi-heartex commented on web/libs/ui/src/tailwind.css:
```diff
@@ -79,3 +79,9 @@ h6 { font-size: 0.67em; }
+
+@layer utilities {
+  .text-shadow-button {
+    text-shadow: 0 1px 1px rgb(var(--black-raw) / 15%);
```
I see we we're using `--black-raw` here - that is not part of the new figma tokens - maybe @ricardoantoniocm can advise on a new standardized token we can use here?

ricardoantoniocm replied:
The color for shadows is `color-neutral-shadow`, in this instance, the raw variant should be used `color-neutral-shadow-raw`.

nick-skriabin committed:
Addressing CS feedback

nick-skriabin committed:
Update color

nick-skriabin committed:
Addressing feedback

nick-skriabin committed:
Remove traces

nick-skriabin committed:
Cleanup

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Biome fixes

nick-skriabin committed:
Update media controls

nick-skriabin committed:
Fix tests

nick-skriabin committed:
Update antd buttons (WIP)

nick-skriabin committed:
Replace antd buttons

nick-skriabin committed:
Update the rest of the buttons

nick-skriabin committed:
Biome

nick-skriabin committed:
Fix relations test

nick-skriabin committed:
Fix e2e and integration tests

nick-skriabin committed:
Fix selectors

nick-skriabin committed:
Biome

nick-skriabin committed:
Fix aria-labels

nick-skriabin committed:
Fix selectors

nick-skriabin committed:
Fix selector

nick-skriabin committed:
One more selectors fix

nick-skriabin committed:
Add proper aria-label to region delete button

nick-skriabin committed:
Restore "delete selected region" behavior

vladimirheartex commented:
/fm sync

nick-skriabin commented:
/docker build
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14595324630)
> Docker image was pushed with the tag `20250422.125810-fb-dia-2142-new-button-ls-81ab6f854`

nick-skriabin committed:
Formatting

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Clarification on data-ignore-uikit

nick-skriabin committed:
Update buttons look

nick-skriabin committed:
Include libs in tailwind compilation process

nick-skriabin commented:
/docker build
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14603362863)
> Docker image was pushed with the tag `20250422.195208-fb-dia-2142-new-button-ls-3d3d5eb7b`

nick-skriabin committed:
Adjust styles

nick-skriabin committed:
Fix button sizes

nick-skriabin committed:
Fix grid image preview buttons

nick-skriabin committed:
Fix save button in the API settings dialog

MihajloHoma commented:
/docker build
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14659268874)
> Docker image was pushed with the tag `20250425.072853-fb-dia-2142-new-button-ls-2f922accc`

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Fix incorrect css

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14751552597)

MihajloHoma committed:
Merge branch 'develop' into 'fb-dia-2142/new-button-lso'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14751552597

MihajloHoma committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14751552597

nick-skriabin committed:
Fix filter line and columns list

borisheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14832954932)

borisheartex committed:
Merge branch 'develop' into 'fb-dia-2142/new-button-lso'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14832954932

borisheartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14832954932

nick-skriabin committed:
Fix button story, remove global shadcn export

nick-skriabin committed:
Re-export Shadcn Badge

nick-skriabin committed:
Add missing exports

nick-skriabin committed:
Move button styles to tailwind class names

nick-skriabin committed:
Modal buttons

nick-skriabin committed:
Update controls look

borisheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14906019023)

borisheartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14906019023

borisheartex committed:
Merge branch 'develop' into 'fb-dia-2142/new-button-lso'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14906019023

nick-skriabin committed:
Adjust icon size to button's size

nick-skriabin committed:
Limit min-width for icon containers

nick-skriabin committed:
Fix border color for active state

nick-skriabin committed:
UI polishing

nick-skriabin committed:
Fix filter button

nick-skriabin committed:
Update button borders

borisheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14965779129)

borisheartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14965779129

nick-skriabin committed:
Adding aria-label attributes

nick-skriabin committed:
More aria-label

nick-skriabin committed:
Adding more aria-label attributes

nick-skriabin committed:
Add aria-label attributes to various buttons for improved accessibility

nick-skriabin committed:
Add aria-label attributes to confirm and info buttons for enhanced accessibility

nick-skriabin committed:
Add aria-label attributes to buttons in Debug, Annotations, and RegionItem components for improved accessibility

nick-skriabin committed:
Update aria-label attributes for buttons in Debug, Annotations, Comments, Taxonomy, and TopBar components to enhance accessibility and clarity

nick-skriabin committed:
Some more aria-label

nick-skriabin committed:
Custom reject button styles

yyassi-heartex commented on label_studio/core/static/css/login.css:
```diff
@@ -207,9 +207,20 @@ html, body {
+.login_page_new_ui form button {
   --button-color: var(--color-neutral-inverted-content);
   border-radius: 4rem;
+  background: var(--sand_900);
```
i see background is being set up here - only to be overridden on line 224

yyassi-heartex replied:
reminder about this

yyassi-heartex replied:
```suggestion
```

nick-skriabin committed:
Cancel skip styles

yyassi-heartex commented on label_studio/core/static/css/login.css:
```diff
@@ -207,9 +207,20 @@ html, body {
+  padding: 10px 15px;
+  border-radius: 100px;
+  box-sizing: border-box;
+  border: 1px solid var(--sand_400);
```
```suggestion
  border: 1px solid var(--color-neutral-border);
```

yyassi-heartex replied:
reminder about this one

yyassi-heartex replied:
going to go ahead and apply this change

nick-skriabin committed:
Update go to import button

yyassi-heartex commented on label_studio/core/static/css/login.css:
```diff
@@ -270,6 +281,10 @@ label{
 }
 
+.login-button {
+  font-size: 16px;
```
```suggestion
  font-size: var(--font-size-body-medium);
```

yyassi-heartex replied:
reminder about this

yyassi-heartex replied:
please double check this one to make sure i suggested the right token

yyassi-heartex commented on label_studio/core/static/css/login.css:
```diff
@@ -207,9 +207,20 @@ html, body {
   --button-color: var(--color-neutral-inverted-content);
   border-radius: 4rem;
+  background: var(--sand_900);
+  color: white;
```
```suggestion
  color: var(--color-neutral-content);
```

yyassi-heartex replied:
replacement with token

yyassi-heartex replied:
can you double check the token i suggested here - not sure if thats the correct one in this usage

yyassi-heartex commented on label_studio/core/static/css/uikit.css:
please do not restore this file

yyassi-heartex commented on web/apps/labelstudio/src/pages/CreateProject/Config/Config.jsx:
```diff
@@ -40,36 +41,28 @@ const Label = ({ label, template, color }) => {
-      <button
+      <span className="flex">
+        <label style={{ background: color }}>
+          <input
```
we're reveting back to native input? is that intended?

yyassi-heartex replied:
reminder about this. i would say this is not the recommended way to go - we need to reuse our components

yyassi-heartex replied:
```suggestion
          <Input
```

yyassi-heartex commented on web/apps/labelstudio/src/pages/CreateProject/Config/Config.jsx:
```diff
@@ -40,36 +41,28 @@ const Label = ({ label, template, color }) => {
-          <path d="M12 12L2 2" />
-        </svg>
-      </button>
+        <IconTrash className="min-w-4 min-h-4 text-negative-content" />
```
looks like we're loosing `Delete label` is that intended?

yyassi-heartex replied:
reminder about this

hlomzik replied:
also the original icon was "cross" because it's in the list and it doesnt' clutter view that much, trash can will be more visible.

nick-skriabin replied:
@yyassi-heartex wdym? the label is there and is not even marked as deleted

@hlomzik good point. will restore

yyassi-heartex commented on web/apps/labelstudio/src/pages/CreateProject/Config/Config.jsx:
```diff
@@ -488,14 +481,7 @@ const Configurator = ({
-            size="compact"
-            style={{ width: 160 }}
-          >
+          <Button type="button" data-leave={true} onClick={onBrowse} size="small" look="outlined" aria-label="Browse templates">
```
width was previously 160px - are we intentionally removing that?

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Add updated button component

nick-skriabin committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15002461003

nick-skriabin committed:
Merge branch 'develop' into 'fb-dia-2175/new-button-lse'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15002461003

nick-skriabin committed:
Use correct button component

nick-skriabin committed:
Update Account Page buttons

nick-skriabin committed:
Add `cn` export

borisheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15015026006)

borisheartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15015026006

borisheartex committed:
Merge branch 'develop' into 'fb-dia-2142/new-button-lso'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15015026006

nick-skriabin committed:
Add waiting state to the button (useful for tests)

nick-skriabin committed:
Add waiting state to the button (useful for tests)

nick-skriabin committed:
Add variant state

nick-skriabin committed:
Add variant state

borisheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15038127514)

borisheartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15038127514

nick-skriabin committed:
Merge branch 'fb-dia-2142/new-button-lso' into fb-dia-2175/new-button-lse

borisheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15064530699)

borisheartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15064530699

nick-skriabin committed:
Restore aria-label for region actions

nick-skriabin committed:
Merge branch 'fb-dia-2142/new-button-lso' into fb-dia-2175/new-button-lse

nick-skriabin committed:
Fix grid preview buttons

nick-skriabin committed:
Grid view buttons

nick-skriabin committed:
Larger icons

nick-skriabin committed:
Remove outline from region actions

nick-skriabin committed:
Gradient button

nick-skriabin committed:
Gradient button focused state

borisheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15107153263)

borisheartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15107153263

nick-skriabin committed:
Fix comment action button

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2175/new-button-lse

nick-skriabin committed:
Biome fixes

nick-skriabin committed:
Biome checks

nick-skriabin committed:
Fix unit tests

nick-skriabin committed:
Restore proper aria-labels

nass600 commented on web/apps/labelstudio/src/pages/Organization/PeoplePage/InviteLink.tsx:
```diff
@@ -84,12 +84,17 @@ const InvitationFooter = () => {
-        <Button look="primary" style={{ width: 170 }} onClick={() => copyText(link!)}>
+        <Button
+          variant={copied ? "positive" : "primary"}
+          style={{ width: 170 }}
```
```suggestion
          className="w-[170px]"
```

nass600 commented on web/apps/labelstudio/src/pages/Settings/MachineLearningSettings/MachineLearningSettings.jsx:
```diff
@@ -172,7 +172,7 @@ export const MachineLearningSettings = () => {
                 <span case="success">Saved!</span>
               </Form.Indicator>
-              <Button type="submit" look="primary" style={{ width: 120 }}>
+              <Button type="submit" look="primary" style={{ width: 120 }} aria-label="Save machine learning settings">
```
```suggestion
              <Button type="submit" look="primary" className="w-[120px]" aria-label="Save machine learning settings">
```

nass600 commented on web/libs/datamanager/src/components/Common/Modal/ModalPopup.jsx:
```diff
@@ -85,12 +84,9 @@ export class Modal extends React.Component {
-                    type="text"
-                    icon={<Icon size="24" color="#617ADA" icon={IconCross} />}
-                  />
+                  <Elem tag={Button} name="close" look="string">
```
Is it possible to remove Elem here and do a full replacement?

nass600 commented on web/apps/labelstudio/src/components/Modal/ModalPopup.jsx:
```diff
@@ -101,7 +101,11 @@ export class Modal extends React.Component {
                   <Elem name="title">{this.state.title}</Elem>
-                  {this.props.allowClose !== false && <Elem tag={Button} name="close" icon={<IconCross />} />}
+                  {this.props.allowClose !== false && (
+                    <Elem tag={Button} name="close" look="outlined">
```
Is it possible to remove Elem here?

nass600 commented on web/apps/labelstudio/src/components/HeidiTips/HeidiTip.tsx:
```diff
@@ -40,7 +40,7 @@ export const HeidiTip: FC<HeidiTipProps> = ({ tip, onDismiss, onLinkClick }) =>
             <Tooltip title="Don't show">
               {/* @ts-ignore-next-line */}
-              <Elem name="dismiss" tag={Button} type="text" onClick={handleClick}>
+              <Elem name="dismiss" tag={Button} look="string" size="small" onClick={handleClick}>
```
Is it possible to remove Elem here?

nass600 commented on web/libs/datamanager/src/components/DataManager/Toolbar/GridWidthButton.jsx:
```diff
@@ -30,20 +30,24 @@ export const GridWidthButton = injector(({ view, gridWidth, size }) => {
-        />
+          aria-label="Decrease number of columns"
+        >
+          <Icon icon={IconMinus} size="12" color="#595959" />
```
Should the color be hardcoded?

nass600 commented on web/libs/datamanager/src/components/DataManager/Toolbar/GridWidthButton.jsx:
```diff
@@ -30,20 +30,24 @@ export const GridWidthButton = injector(({ view, gridWidth, size }) => {
-      </Button.Group>
+          aria-label="Increase number of columns"
+        >
+          <Icon icon={IconPlus} size="12" color="#595959" />
```
Should the color be hardcoded?

nass600 commented on web/libs/editor/src/components/CurrentEntity/HistoryActions.jsx:
```diff
@@ -0,0 +1,51 @@
+  return (
+    <Block name="history-buttons">
+      <Hotkey.Tooltip name="annotation:undo">
+        <Elem
```
Is it possible to replace the Elem?

nass600 commented on web/libs/editor/src/components/CurrentEntity/HistoryActions.jsx:
```diff
@@ -0,0 +1,51 @@
+    <Block name="history-buttons">
+      <Hotkey.Tooltip name="annotation:undo">
+        <Elem
+          tag={Button}
```
Is it possible to replace the Elem?

nass600 commented on web/libs/editor/src/components/CurrentEntity/HistoryActions.jsx:
```diff
@@ -0,0 +1,51 @@
+        />
+      </Hotkey.Tooltip>
+      <Hotkey.Tooltip name="annotation:redo">
+        <Elem
```
Is it possible to replace the Elem?

nass600 commented on web/libs/editor/src/components/Entities/RegionItem.jsx:
```diff
@@ -0,0 +1,164 @@
+          ) : null;
+        })}
+      </Elem>
+      <Elem name="collapse" tag={Button} size="small" type="text" onClick={toggleCollapsed}>
```
Is it possible to replace the Elem?

nass600 commented on web/libs/editor/src/components/Entities/RegionItem.jsx:
```diff
@@ -0,0 +1,164 @@
+          )}
+
+          {item.hideable && (
+            <Elem
```
Is it possible to replace the Elem?

nick-skriabin committed:
Update web/apps/labelstudio/src/pages/Organization/PeoplePage/InviteLink.tsx

Co-authored-by: Ignacio Velazquez <ignacio@humansignal.com>

nick-skriabin committed:
Update web/apps/labelstudio/src/pages/Settings/MachineLearningSettings/MachineLearningSettings.jsx

Co-authored-by: Ignacio Velazquez <ignacio@humansignal.com>

nick-skriabin committed:
Replace Elem buttons with Button

nick-skriabin committed:
Merge branch 'fb-dia-2142/new-button-lso' into fb-dia-2175/new-button-lse

nick-skriabin committed:
Fix integration tests

yyassi-heartex commented on label_studio/core/static/css/uikit.css:
please dont merge this file back in. it was eliminated as part of the token initiative

yyassi-heartex commented on web/apps/labelstudio/src/components/Form/Form.jsx:
```diff
@@ -524,7 +524,7 @@ Form.Builder = React.forwardRef(
         {props.autosubmit !== true && withActions === true && (
           <Form.Actions>
-            <Button type="submit" look="primary" style={{ width: 120 }}>
+            <Button type="submit" className="w-[120px]" aria-label="Submit form">
```
moving away from primary look, is that intended? or is "primary" the new default?

nick-skriabin replied:
new button is primary by default

yyassi-heartex commented on web/apps/labelstudio/src/pages/CreateProject/Config/Config.jsx:
```diff
@@ -40,36 +41,28 @@ const Label = ({ label, template, color }) => {
-        className={configClass.elem("delete-label")}
+        look="string"
+        size="smaller"
+        variant="neutral"
```
i see variant="neutral" is set here, but icon is a trash icon, should this be `variant="negative"` instead?

nick-skriabin replied:
good catch!

yyassi-heartex commented on web/apps/labelstudio/src/pages/CreateProject/Import/Import.jsx:
```diff
@@ -352,19 +351,23 @@ export const ImportPage = ({
+          onSubmit={onLoadURL}
+        >
+          <Input placeholder="Dataset URL" name="url" ref={urlRef} rawClassName="h-[40px]" />
+          <Button type="submit" aria-label="Add URL">
```
look="primary" removed? is that intended or is primary the only default?

nick-skriabin replied:
primary is default

yyassi-heartex commented on web/apps/labelstudio/src/pages/Settings/MachineLearningSettings/MachineLearningList.jsx:
```diff
@@ -83,7 +84,9 @@ const BackendCard = ({ backend, onStartTrain, onEdit, onDelete, onTestRequest })
             }
           >
-            <Button type="link" icon={<IconEllipsis />} style={{ padding: "15px" }} />
+            <Button type="link" style={{ padding: "15px" }} aria-label="Machine learning model options">
```
className would be ideal
```suggestion
            <Button type="link" className="p-15px" aria-label="Machine learning model options">
```

yyassi-heartex commented on web/libs/datamanager/src/components/Common/Form/Form.jsx:
```diff
@@ -537,7 +537,7 @@ Form.Builder = forwardRef(
         {props.autosubmit !== true && withActions === true && (
           <Form.Actions>
-            <Button type="submit" look="primary" style={{ width: 120 }}>
+            <Button type="submit" style={{ width: 120 }}>
```
will type="submit" assume look="primary"?

nick-skriabin replied:
no. primary is just the default look of the button

nick-skriabin committed:
Update integration tests to match selectors

yyassi-heartex commented on web/libs/datamanager/src/components/Common/SDKButtons.jsx:
```diff
@@ -1,12 +1,16 @@
   return sdk.hasHandler(eventName) ? (
     <Button
       {...props}
+      size="small"
```
should we allow size to be passed in as an option? seems like we're required to use small here otherwise
```suggestion
      size={props.size ?? "small"}
```

nick-skriabin replied:
good point

yyassi-heartex commented on web/libs/datamanager/src/components/DataManager/Toolbar/LabelButton.jsx:
```diff
@@ -84,30 +86,30 @@ export const LabelButton = injector(({ store, canLabel, size, target, selectedCo
-          Label Tasks As Displayed
-        </Button>
+            <Button size="small" look="outlined" variant="primary" aria-label={"Toggle open"}>
+              <IconChevronDown />
```
we're losing the visual toggle when the dropdown is opened - not a show stopper but would be nice to preserve it if possible

nick-skriabin replied:
i'm not sure if any of our dropdowns besides this one had it at all. i would prefer a more centralized solution. for now i believe it's good. and you still see the DD, pretty good indication that it's opened :)

yyassi-heartex replied:
Fair enough, i do agree it should be done in a better way than possible at the moment. Perhaps to be addressed with the soon to come drop-down/popover overhaul

yyassi-heartex commented on web/libs/datamanager/src/components/DataManager/Toolbar/RefreshButton.jsx:
```diff
@@ -13,22 +13,17 @@ const injector = inject(({ store }) => {
     <Button
-      size={size}
-      look={needsDataFetch}
+      size="small"
```
size forced to be small instead of passed down, is that intended? if size that gets passed in as a prop is no longer used in this file - lets just clear it out from the props

nick-skriabin committed:
Cleanup

yyassi-heartex commented on web/libs/editor/src/common/Modal/Modal.jsx:
```diff
@@ -62,8 +64,9 @@ export const confirm = ({ okText, onOk, cancelText, onCancel, buttonLook, ...pro
-          size="compact"
+          size="small"
           look={buttonLook ?? "primary"}
+          aria-label="Confirm"
```
```suggestion
          aria-label={okText ?? "OK"}
```

yyassi-heartex commented on web/libs/editor/src/components/BottomBar/Controls.tsx:
```diff
@@ -153,10 +148,14 @@ export const Controls = controlsInjector<{ annotation: MSTAnnotation }>(
     if (buttonsReplacement) {
       // do nothing as all custom buttons are rendered already and we don't need internal buttons
-    } else if (isReview) {
+      return <Block name="controls">{buttons}</Block>;
```
great! but now we can clean up the comment above since its no longer true

nick-skriabin committed:
Update integration tests

nick-skriabin committed:
Update taxonomi per-tag MIG

nick-skriabin committed:
Remove unnecessary comment

yyassi-heartex reviewed (changes requested):
I'm still going through the PR but there are som show stoppers that need to be addressed so i'm requesting changes.
will continue to review the PRs

yyassi-heartex commented on web/libs/editor/src/components/Comments/CommentFormBase.tsx:
```diff
@@ -65,7 +65,7 @@ export const CommentFormBase: FC<CommentFormProps> = observer(
         />
         <Elem tag="div" name="primary-action">
-          <button type="submit">
+          <button type="submit" aria-label="Submit comment">
```
this appears to be a native button. should this be replaced too? or hold off for a follow up?

yyassi-heartex commented on web/libs/editor/src/components/Comments/OldComment/CommentForm.tsx:
```diff
@@ -86,7 +86,7 @@ export const CommentForm: FC<CommentFormProps> = observer(
         />
         <Elem tag="div" name="primary-action">
-          <button type="submit">
+          <button type="submit" aria-label="Submit comment">
```
native button here too

yyassi-heartex commented on web/libs/editor/src/components/Controls/Controls.jsx:
```diff
@@ -4,8 +4,7 @@ import { CheckCircleOutlined, CheckOutlined } from "@ant-design/icons";
 import styles from "./Controls.module.scss";
-import { Button } from "../../common/Button/Button";
-import { Tooltip } from "@humansignal/ui";
+import { Button, Tooltip } from "@humansignal/ui";
```
if the only usage of Tooltip here is around the Button, lets switch it to use the Button tooltip prop so we can remove the unneeded import

yyassi-heartex commented on web/libs/editor/src/components/CurrentEntity/Controls.jsx:
```diff
@@ -0,0 +1,111 @@
+/**
+ * @deprecated This component is not used anywhere.
```
this is confusing, github is reporting a new file but its deprcated? feels like a file rename. is that what happened or am I missing something?

nick-skriabin replied:
i just accidentally re-intoduced some files that were removed in upstream. removed

yyassi-heartex commented on web/libs/editor/src/components/Filter/Filter.tsx:
this feels like its unrelated to Button - what happened here?

nick-skriabin replied:
same as above

yyassi-heartex commented on web/libs/editor/src/components/Entity/Entity.jsx:
this feels like its unrelated to Button - what happened here?

nick-skriabin replied:
same as above

yyassi-heartex commented on web/libs/editor/src/components/Entities/Entities.jsx:
this feels like its unrelated to Button - what happened here?

nick-skriabin replied:
same as above

yyassi-heartex commented on web/libs/editor/src/components/Entities/LabelItem.jsx:
this feels like its unrelated to Button - what happened here?

nick-skriabin replied:
same as above

yyassi-heartex commented on web/libs/editor/src/components/Entities/RegionItem.jsx:
this feels like its unrelated to Button - what happened here?

nick-skriabin replied:
same as above

yyassi-heartex commented on web/libs/editor/src/components/Taxonomy/Taxonomy.tsx:
```diff
@@ -435,7 +435,7 @@ const TaxonomyDropdown = ({ show, flatten, items, dropdownRef, isEditable }: Tax
           ) : isEditable ? (
             <div className={styles.taxonomy__add}>
-              <button type="button" onClick={addInside}>
+              <button type="button" onClick={addInside} aria-label="Add new label">
```
native button here, would it be possible to switch it out for new Button?

yyassi-heartex commented on web/libs/editor/src/components/Tools/SliderDropDown.jsx:
```diff
@@ -1,7 +1,7 @@
+import { Dropdown, Menu, Slider } from "antd";
+import { Button, IconRefresh } from "@humansignal/ui";
 import { observer } from "mobx-react";
-import { ReloadOutlined } from "@ant-design/icons";
```
wow i didn't know we were using antd icons too! glad you removed this!

yyassi-heartex commented on web/libs/editor/src/components/TopBar/Actions.jsx:
```diff
@@ -113,7 +115,8 @@ export const Actions = ({ store }) => {
         <Button
           icon={<IconInfo style={{ width: 16, height: 16 }} />}
           primary={store.showingDescription}
```
is primary prop still a valid option?

yyassi-heartex commented on web/libs/editor/src/tags/control/TextArea/TextArea.jsx:
```diff
@@ -420,7 +420,7 @@ const HtxTextArea = observer(({ item }) => {
             {showAddButton && (
               <Form.Item>
-                <Button style={{ marginTop: "10px" }} type="primary" htmlType="submit">
+                <Button size="small" style={{ marginTop: "10px" }} type="primary" htmlType="submit">
```
lets use a tailwind class here instead of inline class

yyassi-heartex commented on web/libs/editor/src/tags/object/Paragraphs/HtxParagraphs.jsx:
```diff
@@ -487,6 +487,8 @@ class HtxParagraphsView extends Component {
       const container = this.myRef.current;
       const mainContentView = document.querySelector(this.mainContentSelector);
+
+      if (!mainContentView) return;
```
this doesn't seem to be Button related, does this address an existing issue? how does `mainContentView` get into a state where it's falsey?

nick-skriabin replied:
not intentional, re-introed when resolving conflicts

yyassi-heartex commented on web/libs/ui/src/assets/icons/interpolation-disabled.svg:
```diff
@@ -1,6 +1,6 @@
 <g opacity="0.2">
-<path fill-rule="evenodd" clip-rule="evenodd" d="M19 7C18.4477 7 18 7.44772 18 8V10.9998H7C6.44772 10.9998 6 11.4475 6 11.9998C6 12.552 6.44772 12.9998 7 12.9998H18V16C18 16.5523 18.4477 17 19 17C19.5523 17 20 16.5523 20 16V12.9998H23C23.5523 12.9998 24 12.552 24 11.9998C24 11.4475 23.5523 10.9998 23 10.9998H20V8C20 7.44772 19.5523 7 19 7Z" fill="black"/>
-<path d="M2.82843 14.8284C1.49509 13.4951 0.828427 12.8284 0.828427 12C0.828427 11.1716 1.49509 10.5049 2.82843 9.17157L3.17157 8.82843C4.50491 7.49509 5.17157 6.82843 6 6.82843C6.82843 6.82843 7.49509 7.49509 8.82843 8.82843L9.17157 9.17157C10.5049 10.5049 11.1716 11.1716 11.1716 12C11.1716 12.8284 10.5049 13.4951 9.17157 14.8284L8.82843 15.1716C7.49509 16.5049 6.82843 17.1716 6 17.1716C5.17157  ...
+<path fill-rule="evenodd" clip-rule="evenodd" d="M19 7C18.4477 7 18 7.44772 18 8V10.9998H7C6.44772 10.9998 6 11.4475 6 11.9998C6 12.552 6.44772 12.9998 7 12.9998H18V16C18 16.5523 18.4477 17 19 17C19.5523 17 20 16.5523 20 16V12.9998H23C23.5523 12.9998 24 12.552 24 11.9998C24 11.4475 23.5523 10.9998 23 10.9998H20V8C20 7.44772 19.5523 7 19 7Z" fill="currentColor"/>
```
great catch! this would cause an issue in darkmode

yyassi-heartex commented on web/libs/ui/src/assets/icons/keypoint-disabled.svg:
```diff
@@ -1,6 +1,6 @@
 <svg width="36" height="36" viewBox="0 0 36 36" fill="none" xmlns="http://www.w3.org/2000/svg">
 <g opacity="0.2">
-<rect x="8.10049" y="18" width="14" height="14" rx="2" transform="rotate(-45 8.10049 18)" fill="black"/>
+<rect x="8.10049" y="18" width="14" height="14" rx="2" transform="rotate(-45 8.10049 18)" fill="currentColor"/>
```
@ricardoantoniocm this icon uses 2 colors, with swap out of black towards currentColor this could have issues in light mode with white on `currentColor` applied. can you suggest an updated icon that would work in both light and dark mode for this? we should look at addressing this in a separate ticket as its not directly related to Button.
@nick-skriabin great catch on this - but I'm concerned that we are addressing too much in this PR

yyassi-heartex reviewed (changes requested):
I like the direction of this but there are alot of changes, some of which would have been better suited to be in a separate PR to help with delivery speed from implementation to review. there are 2 more PRs to go

nick-skriabin committed:
Remove old code (regression from develop)

nick-skriabin committed:
Replace native buttons, remove tooltip component

nick-skriabin committed:
Replace native button

nick-skriabin committed:
Remove `primary` prop

nick-skriabin committed:
Use TW class

nick-skriabin committed:
Remove conflict resolution artifacts

nick-skriabin committed:
Merge branch 'fb-dia-2142/new-button-lso' into fb-dia-2175/new-button-lse

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2175/new-button-lse

hlomzik commented on web/libs/datamanager/src/components/DataManager/Toolbar/LabelButton.jsx:
```diff
@@ -84,30 +86,30 @@ export const LabelButton = injector(({ store, canLabel, size, target, selectedCo
-        <div style={{ display: "flex" }}>
+        <ButtonGroup>
           <Button
-            size={size}
```
Why did you remove `size` from here? I'm not sure it's actually used, but all `instruments` in DM's Toolbar use `size` at least in the code.

nick-skriabin replied:
restored

yyassi-heartex committed:
Update label_studio/core/static/css/login.css

hlomzik commented on web/libs/editor/src/core/Hotkey.ts:
```diff
@@ -422,7 +422,7 @@ Hotkey.setScope = (scope: string) => {
 Hotkey.Tooltip = inject("store")(
   observer(({ store, name, children, ...props }: any) => {
-    const hotkey = Hotkey.keymap[name as string];
+    const hotkey = Hotkey.keymap[name as keyof typeof Hotkey.keymap];
```
```suggestion
    const hotkey = Hotkey.keymap[name as HotkeyList];
```

hlomzik commented on web/libs/editor/src/stores/CustomButton.ts:
```diff
@@ -13,9 +13,11 @@ export const CustomButton = types
+    variant: types.maybe(
+      types.enumeration(["primary", "neutral", "positive", "negative", "warning", "inverted"] as const),
     ),
+    look: types.maybe(types.enumeration(["filled", "outlined", "string"])),
```
This change should be reflected in all the usages of such buttons

nick-skriabin replied:
done in LSE PR

hlomzik commented on web/libs/editor/src/components/TopBar/TopBar.jsx:
```diff
@@ -31,44 +30,37 @@ export const TopBar = observer(({ store }) => {
+            </Button>
           )}
           {store.hasInterface("annotations:add-new") && (
             <Tooltip title="Create a new annotation" style={{ "--offset-x": "11px" }}>
```
Does it make sense to move this tooltip inside the button as well? That will be consistent. And then you can just remove tooltip component from this file.

hlomzik commented on web/libs/editor/src/components/SidePanels/OutlinerPanel/ViewControls.tsx:
```diff
@@ -195,26 +195,24 @@ const Grouping = <T extends string>({
   }, [value, optionsList, readableValue, direction, onChange]);
 
   // mods are already set in the button from type, so use it only in new UI
```
```suggestion
```

nick-skriabin committed:
Restore size prop for instruments

nick-skriabin committed:
Update web/libs/editor/src/core/Hotkey.ts

Co-authored-by: Andrew <hlomzik@gmail.com>

nick-skriabin committed:
Remove unnecessary tooltip

nick-skriabin committed:
Update web/libs/editor/src/components/SidePanels/OutlinerPanel/ViewControls.tsx

Co-authored-by: Andrew <hlomzik@gmail.com>

hlomzik commented on web/libs/editor/src/components/BottomBar/buttons.tsx:
```diff
@@ -94,49 +92,49 @@ type SkipButtonProps = {
-            const action = () => store.skipTask({});
-            const selected = store.annotationStore?.selected;
+      <Button
+        aria-label="skip-task"
```
These buttons are used in a list, so should have a `key` here to not trigger react warnings
```suggestion
      <Button
        key="skip"
        aria-label="skip-task"
```

hlomzik commented on web/libs/editor/src/components/BottomBar/buttons.tsx:
```diff
@@ -50,22 +49,20 @@ type AcceptButtonProps = {
-          look="primary"
-          onClick={async () => {
-            const selected = store.annotationStore?.selected;
+      <Button
+        aria-label="accept-annotation"
```
These buttons are used in a list, so should have a `key` here to not trigger react warnings
```suggestion
      <Button
        key="accept"
        aria-label="accept-annotation"
```

hlomzik commented on web/libs/editor/src/common/Pagination/Pagination.tsx:
```diff
@@ -173,13 +172,9 @@ const NavigationButton: FC<NavigationButtonProps> = ({ mod, disabled, hotkey, on
-    <Hotkey.Tooltip name={hotkey}>
+  return (
+    <WithHotkey binging={hotkey as HotkeyList}>
       <Elem name="btn" mod={buttonMod} onClick={actionHandler} />
```
hmmm, not `Button`?

nick-skriabin replied:
pagination is a relatively complex component and it haven't been using buttons in the first place (why - is another question, but I don't remember). it's out of scope to transition components from non-button to button. we will be migrating this one anyway, so it can be done there

hlomzik commented on web/libs/datamanager/src/components/MainView/GridView/ImagePreview.tsx:
```diff
@@ -195,7 +196,6 @@ const ImagePreview = observer(({ task, field }: ImagePreviewProps) => {
     ? {
         maxWidth: "100%",
         maxHeight: "100%",
-        transform: `translate(${offset.x}px, ${offset.y}px) scale(${scale})`,
```
hmmm, why do you remove this line? it's crucial for images being interactive in this preview

nick-skriabin replied:
probably just an accident. restored

hlomzik commented on web/libs/datamanager/src/components/Filters/Filters.jsx:
```diff
@@ -70,20 +69,20 @@ export const Filters = injector(({ views, currentView, filters }) => {
-              style={{ display: "inline-flex", alignItems: "center", padding: 0, width: "var(--button-height)" }}
-              icon={<Icon icon={IconChevronRight} size={18} />}
-            />
+            <Button look="string" type="link" size="small" about="Pin to sidebar" onClick={() => views.expandFilters()}>
```
I believe `about` was a mistake in both old and new versions?
```suggestion
            <Button look="string" type="link" size="small" tooltip="Pin to sidebar" onClick={() => views.expandFilters()}>
```

hlomzik replied:
Also "Unpin sidebar" lost this field in `FilterSidebar`

hlomzik replied:
plus we also need `aria-label` here

hlomzik commented on web/libs/datamanager/src/components/Filters/FilterLine/FilterLine.jsx:
```diff
@@ -68,15 +67,15 @@ export const FilterLine = observer(({ filter, availableFilters, index, view, sid
-          icon={<Icon icon={IconClose} size={12} />}
-        />
+        >
+          <IconTrash className="!h-4 !w-4" />
```
You use `size` in all other icons, so let's change it for consistency?
```suggestion
          <IconTrash size={12} />
```

hlomzik commented on web/libs/datamanager/src/components/Common/Modal/Modal.jsx:
```diff
@@ -62,8 +64,8 @@ export const confirm = ({ okText, onOk, cancelText, onCancel, buttonLook, ...pro
             modal.close();
           }}
-          size="compact"
-          look={buttonLook ?? "primary"}
```
are we loosing look for this and next button?

hlomzik replied:
I see in another comment that "primary" was just a default look for a button. But what about `buttonLook`? If we really don't need it then better to remove it as a property. But that could be done later, the main thing is that we don't loose button states here.

nick-skriabin replied:
we haven't lost it actually. it's used in the `variant` property now. i just decided to leave the original name of the property

hlomzik commented on web/libs/datamanager/src/components/Common/Dropdown/DropdownComponent.jsx:
```diff
@@ -153,10 +153,11 @@ export const Dropdown = React.forwardRef(({ animated = true, visible = false, ..
     ...(offset ?? {}),
     zIndex: 1000 + dropdownIndex,
   };
+  console.log({ rawClassName });
```
```suggestion
```

hlomzik commented on web/apps/labelstudio/src/pages/Projects/Projects.jsx:
```diff
@@ -160,7 +160,7 @@ ProjectsPage.routes = ({ store }) => [
   if (!showButton) return null;
   return (
-    <Button onClick={openModal} look="primary" size="compact">
+    <Button onClick={openModal} look="primary" size="small" aria-label="Create new project">
```
I think we don't have `look="primary"`, right?

hlomzik commented on web/apps/labelstudio/src/pages/Projects/ProjectsList.jsx:
```diff
@@ -39,12 +39,12 @@ export const EmptyProjectsList = ({ openModal }) => {
       </Elem>
       <p>Create one and start labeling your data.</p>
-      <Elem name="action" tag={Button} onClick={openModal} look="primary">
+      <Button onClick={openModal} className="my-8" aria-label="Create new project">
```
"primary" look is missing here

nick-skriabin replied:
buttons are primary by default

hlomzik commented on web/apps/labelstudio/src/pages/Projects/ProjectsList.jsx:
```diff
@@ -39,12 +39,12 @@ export const EmptyProjectsList = ({ openModal }) => {
       <Elem name="heidi" tag="img" src={absoluteURL("/static/images/opossum_looking.png")} />
       <Elem name="header" tag="h1">
-        Heidi doesn’t see any projects here!
+        Heidi doesn't see any projects here!
```
I don't think we need to change nice apostrophe in a regular text
```suggestion
        Heidi doesn’t see any projects here!
```

hlomzik commented on web/apps/labelstudio/src/pages/Projects/ProjectsList.jsx:
```diff
@@ -119,7 +121,7 @@ const ProjectCard = ({ project }) => {
         <Elem name="description">{project.description}</Elem>
         <Elem name="info">
-          <Elem name="created-date">{format(new Date(project.created_at), "dd MMM ’yy, HH:mm")}</Elem>
+          <Elem name="created-date">{format(new Date(project.created_at), "dd MMM 'yy, HH:mm")}</Elem>
```
not sure here though, but this change is also out of scope of this PR

nick-skriabin replied:
it's probably the formatter. i definitely didn't touch it

hlomzik commented on web/apps/labelstudio/src/components/Modal/Modal.jsx:
```diff
@@ -113,6 +116,7 @@ export const info = ({ okText, onOkPress, ...props }) => {
           }}
           look="primary"
```
Wrong `look`?

nick-skriabin committed:
Update web/libs/editor/src/components/BottomBar/buttons.tsx

Co-authored-by: Andrew <hlomzik@gmail.com>

nick-skriabin committed:
Update web/libs/editor/src/components/BottomBar/buttons.tsx

Co-authored-by: Andrew <hlomzik@gmail.com>

nick-skriabin committed:
Update web/apps/labelstudio/src/pages/Projects/ProjectsList.jsx

Co-authored-by: Andrew <hlomzik@gmail.com>

nick-skriabin committed:
Various tiny fixes

nick-skriabin committed:
Restore transform

nick-skriabin committed:
Remove excess background property

nick-skriabin committed:
Use variable for font size

nick-skriabin committed:
Use token for text color

nick-skriabin committed:
Fine tuning

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Update sample remove button

nick-skriabin committed:
Dark mode fixed

nick-skriabin committed:
Fix border radius on login page inputs

nick-skriabin committed:
Fix modal close button

nick-skriabin committed:
Fix danger zone buttons

nick-skriabin committed:
Fix machine learning ellipsis button

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Fix config delete label button

nick-skriabin committed:
Fix grid view toggle and buttons

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2142/new-button-lso

nick-skriabin commented:
/fmt
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15591774620)

nick-skriabin committed:
Apply pre-commit linters

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15591774620

yyassi-heartex reviewed (approved):
approving assuming all checks are green and LSE selenium passes

nick-skriabin committed:
Fix gradient button animation

nick-skriabin committed:
Merge branch 'fb-dia-2175/new-button-lse' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2142/new-button-lso

yyassi-heartex commented on web/libs/ui/src/tailwind.config.js:
```diff
@@ -2,15 +2,7 @@ import tokens from "./tokens/tokens";
-    "./libs/ui/src/**/*.{js,jsx,ts,tsx}",
-    "./libs/storybook/**/*.{js,jsx,ts,tsx}",
-  ],
+  content: ["./apps/**/*.{js,jsx,ts,tsx}", "./libs/**/*.{js,jsx,ts,tsx}"],
```
discussing with @bmartel - if the e2e/integration test environments are getting timed out its likely because of picking up too many things here. please take another look

yyassi-heartex replied:
solution maybe to just revert to what we had before to make sure we dont get extra/unexpected stuff

nick-skriabin committed:
Tailwind config develop

nick-skriabin committed:
Remove outline from the modal close button

nick-skriabin committed:
Add aria label to close modal button

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2175/new-button-lse

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2175/new-button-lse

nick-skriabin committed:
Merge branch 'fb-dia-2175/new-button-lse' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Add aria-label to the actions button

nick-skriabin committed:
Add aria-label to task action buttons

nick-skriabin committed:
Fix aria labels

nick-skriabin committed:
Remove console.log

nick-skriabin committed:
Fix modal window not closing

nick-skriabin committed:
Fix modal not closing

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2175/new-button-lse

borisheartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16025397476

borisheartex committed:
Merge branch 'develop' into 'fb-dia-2175/new-button-lse'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16025397476

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Fix icon import

nick-skriabin committed:
Fix filter icons

borisheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/16068319129)

borisheartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16068319129

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Fix tests

nick-skriabin committed:
Custom hotkeys button

nick-skriabin committed:
Fix references

nick-skriabin commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/16120193272)
> Successfully merged:  2 files changed, 12 insertions(+), 810 deletions(-)

nick-skriabin committed:
Merge branch 'develop' into 'fb-dia-2142/new-button-lso'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16120193272

nick-skriabin committed:
Tasks

nick-skriabin commented:
/fmt
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/16120785427)

nick-skriabin committed:
Apply pre-commit linters

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16120785427

nick-skriabin committed:
Fix tests

nick-skriabin committed:
Fix test

nick-skriabin committed:
Enable tasks

nick-skriabin committed:
Restore screenshot comparison

nick-skriabin committed:
Fix controls

nick-skriabin committed:
Fix tests

nick-skriabin committed:
Fix last test

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2142/new-button-lso

nick-skriabin committed:
Fix audio paragraphs test

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2142/new-button-lso
